internal/course/domain/course_repository.go

```diff
@@ -1,11 +1,14 @@
 package domain
 
-import "github.com/google/uuid"
+import (
+	"github.com/google/uuid"
+	"github.com/sumelms/microservice-course/internal/shared"
```

Comment:
Here we are creating an external dependency for the domain, but I don't think that this is a problem. There is nothing on DDD that says that we can't do that. Typically, however, you will always desire to have your domain as independent as possible. But, it is just a comment... in this specific case, I don't think that it is a problem.

Reply:
https://github.com/sumelms/microservice-course/pull/26/commits/55dd95b48f589a3da02aee513c163518f3e84f1a